asmcomp/emitaux.ml

```diff
@@ -178,6 +183,12 @@ let emit_frames a =
       Label_table.add debuginfos key lbl;
       lbl
   in
+  let efa_16_checked n =
+    assert (n >= 0);
+    if n < 0x1_0000
+    then a.efa_16 n
+    else raise (Error(Stack_frame_too_large n))
```

Comment:
It is strange that the error precisely mentions stack frames, while this function could be used for other things by emitaux or a specific code emitter. One way to avoid this would be to have a generic `Out_of_range` exception raised here, and have the user code in `emit_frame` catch it and raise this contextual exception. (Two reraises for the price of one!)

Reply:
The `efa_16_checked` function is local to `emit_frame`, so we know all the call sites, and they are all about stack frame characteristics (size, number of roots, offsets of roots).  I don't think more generality is warranted.